core/src/main/scala/io/delta/tables/DeltaColumnBuilder.scala

```diff
@@ -130,6 +130,9 @@ class DeltaColumnBuilder private[tables](
     if (dataType == null) {
       throw DeltaErrors.analysisException(s"The data type of the column $colName is not provided")
     }
+    if (dataType == DataTypes.NullType) {
```

Comment:
Would be possible to use SchemaUtils.findNullTypeColumn to validate it? It is already used for merge

Reply:
Of course!  Let me have a crack.  I will push updates when complete

Reply:
@felipepessoto I have been playing around with this.  I don't think the SchemaUtils function, as it is implemented today, checks for the Null dataType.  I think it is looking for when the dataType is actuall "null", but it is not catching when the value is "DataTypes.Null".

Should I be looking at enhancing the SchemaUtils.findNullTypeColumn to include checking for DataTypes.Null as well?

Reply:
I think the usage would be similar to https://github.com/delta-io/delta/blob/24c025128612a4ae02d0ad958621f928cda9a3ec/core/src/main/scala/org/apache/spark/sql/delta/commands/MergeIntoCommand.scala#L326

But you won't be able to do it here, since you need the schema.

Reply:
I did try that.  My issue is that function isn't handling when the type is DataTypes.NullType... it only handles NullValue... which is slightly different.  If I don't change the implementation of the utils function, it will not handle when the value has been defined as DataTypes.NullType

Reply:
I see. In that case I wouldn't change the existing findNullTypeColumn, to avoid breaking changes. I'd leave as is now